basex-core/src/main/java/org/basex/core/locks/DBLocking.java

```diff
@@ -138,7 +138,9 @@ public void acquire(final Proc pr, final StringList read, final StringList write
       }
       // global read locking
       if(read == null) {
-        while(localWriters > 0) {
+        while(localWriters > 0 &&
+            // We're the only writer, allow global read lock anyway
+            !(1 == localWriters && !(null == write || write.isEmpty()))) {
```

Comment:
Yes, that's it.

**Loop (wait for resources to be released):**
* `localWriters` > `1` (there is somebody else writing somewhere)
* `localWriters` = `1`, global write (we did not specify a local write lock, cannot be us)
* `localWriters` = `1`, no local writes (we do not write, cannot be us)

**Stop (resources are available, continue:**
* `localWriters` = `0` (no writers at all, fine anyway)
* `localWriters` = `1`, local writes (there is exactly one writer, and we are this one)